tenders/hvt/hvt_freebsd.c

```diff
@@ -158,13 +158,15 @@ int hvt_guest_mprotect(void *t_arg, uint64_t addr_start, uint64_t addr_end,
     /*
      * Host-side page protections:
      *
-     * Ensure that guest-executable pages are not also executable in the host.
+     * Ensure that guest-executable pages are not also executable but are
+     * readable in the host.
      *
      * Guest-side page protections:
      *
      * Manipulating guest-side (EPT) mappings is currently not supported by
      * FreeBSD vmm, so there is nothing more we can do.
      */
     prot &= ~(PROT_EXEC);
+    prot |= PROT_READ;
```

Comment:
This (and the other host variants) are a bit hard to read now. I'd make it explicit that we're "squashing" `PROT_EXEC` into `PROT_READ` on the host side, and nothing else:

```c
if (prot & PROT_EXEC) {
    prot &= ~(PROT_EXEC);
    prot |= PROT_READ;
}
```

This makes the intent crystal clear, i.e.

- PROT_READ: (no change)
- PROT_WRITE: (no change)
- PROT_EXEC: -> PROT_READ

What do you think?

Reply:
At first just reading this, i was unsure, but i thought i would implement it and then reread the code, and its much more readable.